src/company_gui.cpp

```diff
@@ -2674,11 +2679,10 @@ struct CompanyWindow : Window
 		}
 	}
 
-	void OnHundredthTick() override
-	{
-		/* redraw the window every now and then */
+	/** Redraw the window on a regular interval. */
+	IntervalTimer<TimerWindow> redraw_interval = {std::chrono::seconds(3), [this](auto) {
```

Comment:
Does the actual time value ever vary? Is it likely to? Perhaps it could be a template parameter instead?

Reply:
There is at least one case where the timer is scaled by scaled font size.

Reply:
There is at least one instance of `std::chrono::seconds(3 * _settings_client.gui.errmsg_duration)`, and I suspect there will be a few more over time.

So yeah, sadly, no template parameter :(